test/datetime/create.test.js

```diff
@@ -883,6 +883,12 @@ test("DateTime.fromObject takes a undefined to mean {}", () => {
   expect(res.year).toBe(new Date().getFullYear());
 });
 
+test("DateTime.fromObject respects `referenceDate`", () => {
+  const res = DateTime.fromObject(undefined, { referenceDate: { day: 10 } });
+  expect(res.year).toBe(new Date().getFullYear());
+  expect(res.day).toBe(10);
+});
+
```

Comment:
I think it would be good to also add tests for:
- Calling `fromObject` with actual data and check that the data is "merged in".
- Checking that `referenceDate` is only used for units higher than the highest unit in the source object (i.e. `fromObject({ day: 10 })` reads _only_ `year` and `month` from the reference date.
- Some tests for methods that indirectly call `fromObject` (like `fromISO`) with `referenceDate`.